cloudify_rest_client/secrets_providers.py

```diff
@@ -100,6 +105,46 @@ def create(
 
         return SecretsProvider(response)
 
+    def update(
+            self,
+            name,
+            _type,
+            connection_parameters,
+            visibility=VisibilityState.TENANT,
+    ):
+        """
+        Update already exists Secrets Provider.
+
+        :param name: Secrets Provider name
+        :type name: str
+        :param _type: Secrets Provider type
+        :type _type: str
+        :param connection_parameters: Secrets Provider connection parameters
+        :type connection_parameters: dict
+        :param visibility: The visibility of the secret, can be 'private',
+                           'tenant' or 'global'
+        :type visibility: str
+        :returns: New secrets provider metadata
+        :rtype: Dict[str]
+        """
+        data = {
+            'name': name,
+            'type': _type,
+            'connection_parameters': connection_parameters,
+            'visibility': visibility,
+        }
+        data = dict((k, v) for k, v in data.items() if v is not None)
+
+        response = self.api.patch(f'/secrets-providers/{name}', data=data)
+
+        return SecretsProvider(response)
+
+    def delete(self, name):
+        """
+        Deletes single Secrets Providers.
```

Comment:
```suggestion
        Delete a Secrets Provider.
```
tbh I'm not sure if we use the `Deletes` for or the `Delete` form. But we do have `Update` here, so might as well do `Delete` as well.